YourEasyRent/DataBase/TelegramRepository.cs

```diff
@@ -0,0 +1,31 @@
+﻿using MongoDB;
+using MongoDB.Driver;
+using Telegram.Bot;
+using YourEasyRent.DataBase.Interfaces;
+using YourEasyRent.Entities;
+
+namespace YourEasyRent.DataBase
+{
+    public class TelegramRepository : ITelegramRepository
+    {
+
+        private readonly IMongoCollection<Product> _product;  // объявляет приватное поле _product в текущем классе TelegramRepository, которое будет использоваться для внедрения зависимости ITelegramRepository. Это поле будет доступно для использования в методах и конструкторах класса
+
+        public TelegramRepository(DataBaseConfig configuration, IMongoClient client) // объявляем конструктор. Конструктор принимает два аргумента: DataBaseConfig configuration и IMongoClient client. Класс DataBaseConfig используется для передачи конфигурационных данных, а IMongoClient представляет клиент MongoDB, который используется для установления соединения с базой данных.
+        {
+            var database = client.GetDatabase(configuration.DataBaseName);
+            _product = database.GetCollection<Product>(configuration.CollectionName);
+        }
+        public async Task<IEnumerable<Product>> GetProductsByBrandAndPrice(string brand, decimal price)
+        {
+            var filter = Builders<Product>.Filter.And
```

Comment:
Nice job, but there is easier way to do the same, you ca see it in my PR, but my way does not work for every case.